app/screens/redeem-lnurl-withdrawal-screen/redeem-bitcoin-result-screen.tsx

```diff
@@ -0,0 +1,388 @@
+  })
+  const usdAmount = satAmountInUsd
+  const usdAmountInSats = Math.round(
+    convertCurrencyAmount({
+      amount: usdAmount ?? 0,
+      from: "USD",
+      to: "BTC",
+    }),
+  )
+
+  const [err, setErr] = useState("")
+  const [withdrawalInvoice, setInvoice] = useState<LnInvoice | null>(null)
+
+  const [memo] = useState(defaultDescription)
+
+  const { lnUpdate } = useSubscriptionUpdates()
+  const invoicePaid =
+    lnUpdate?.paymentHash === withdrawalInvoice?.paymentHash &&
+    lnUpdate?.status === "PAID"
+
+  const [lnInvoiceCreate] = useLnInvoiceCreateMutation()
+  const [lnUsdInvoiceCreate] = useLnUsdInvoiceCreateMutation()
+
+  const createWithdrawRequestInvoice = useCallback(
+    async ({ satAmount, memo }) => {
+      setInvoice(null)
+      try {
+        if (type === TYPE_LIGHTNING_BTC) {
+          logGeneratePaymentRequest({
+            paymentType: "lightning",
+            hasAmount: true,
+            receivingWallet: WalletCurrency.Btc,
+          })
+          const {
+            data: {
+              lnInvoiceCreate: { invoice, errors },
+            },
+          } = await lnInvoiceCreate({
+            variables: {
+              input: { walletId, amount: satAmount, memo },
+            },
+          })
+          if (errors && errors.length !== 0) {
+            console.error(errors, "error with lnInvoiceCreate")
+            setErr(LL.RedeemBitcoinScreen.error())
+            return
+          }
+
+          setInvoice(invoice)
+        } else {
+          logGeneratePaymentRequest({
+            paymentType: "lightning",
+            hasAmount: true,
+            receivingWallet: WalletCurrency.Usd,
+          })
+          const {
+            data: {
+              lnUsdInvoiceCreate: { invoice, errors },
+            },
+          } = await lnUsdInvoiceCreate({
+            variables: {
+              input: { walletId, amount: satAmountInUsd * 100, memo },
+            },
+          })
+
+          if (errors && errors.length !== 0) {
+            console.error(errors, "error with lnInvoiceCreate")
+            setErr(LL.ReceiveBitcoinScreen.error())
+            return
+          }
+          setInvoice(invoice)
+        }
+      } catch (err) {
+        console.error(err, "error with AddInvoice")
+        setErr(`${err}`)
+        throw err
+      }
+    },
+    [lnInvoiceCreate, lnUsdInvoiceCreate, LL],
+  )
+
+  const submitLNURLWithdrawRequest = async (generatedInvoice) => {
+    const url = `${callback}${callback.includes("?") ? "&" : "?"}k1=${k1}&pr=${
+      generatedInvoice.paymentRequest
+    }`
+
+    const result = await fetch(url)
+
+    if (result.ok) {
+      const lnurlResponse = await result.json()
+      if (lnurlResponse?.status?.toLowerCase() === "ok") {
+        // TODO: Set processing payment
+      } else {
+        console.error(lnurlResponse, "error with redeeming")
+        // TODO: Set failed payment
+        setErr(LL.RedeemBitcoinScreen.redeemingError())
+      }
```

Comment:
is this still in progress? else please update the condition and remove the else block